mis_builder/models/aep.py

```diff
@@ -0,0 +1,205 @@
+          strict minimum number of queries to the database (for each period,
+          one query per domain and mode);
+        * it queries using the orm read_group which reduces to a query with
+          sum on debit and credit and group by on account_id (note: it seems
+          the orm then does one query per account to fetch the account
+          name...);
+        * additionally, one query per view/consolidation account is done to
+          discover the children accounts.
+    """
+
+    ACC_RE = re.compile(r"(?P<field>\bbal|\bcrd|\bdeb)"
+                        r"(?P<mode>[pise])?"
+                        r"(?P<accounts>_[0-9]+|\[.*?\])"
+                        r"(?P<domain>\[.*?\])?")
+
+    def __init__(self, env):
+        self.env = env
+        self._map = defaultdict(set)  # {(domain, mode): set(account_ids)}
+        self._account_ids_by_code = defaultdict(set)
+
+    def _load_account_codes(self, account_codes, account_domain):
+        account_model = self.env['account.account']
+        # TODO: account_obj is necessary because _get_children_and_consol
+        #       does not work in new API?
+        account_obj = self.env.registry('account.account')
+        exact_codes = set()
+        like_codes = set()
+        for account_code in account_codes:
+            if account_code in self._account_ids_by_code:
+                continue
+            if '%' in account_code:
+                like_codes.add(account_code)
+            else:
+                exact_codes.add(account_code)
+        for account in account_model.\
+                search([('code', 'in', list(exact_codes))] + account_domain):
+            if account.type in ('view', 'consolidation'):
+                self._account_ids_by_code[account.code].update(
+                    account_obj._get_children_and_consol(
+                        self.env.cr, self.env.uid,
+                        [account.id],
+                        self.env.context))
+            else:
+                self._account_ids_by_code[account.code].add(account.id)
+        for like_code in like_codes:
+            for account in account_model.\
+                    search([('code', 'like', like_code)] + account_domain):
+                if account.type in ('view', 'consolidation'):
+                    self._account_ids_by_code[like_code].update(
+                        account_obj._get_children_and_consol(
+                            self.env.cr, self.env.uid,
+                            [account.id],
+                            self.env.context))
+                else:
+                    self._account_ids_by_code[like_code].add(account.id)
+
+    def _parse_mo(self, mo):
+        """Split a match object corresponding to an accounting variable
+
+        Returns field, mode, [account codes], [domain expression].
+        """
+        field, mode, account_codes, domain = mo.groups()
+        if not mode:
+            mode = 'p'
+        elif mode == 's':
+            mode = 'e'
```

Comment:
mode 's' is undocumented !
And I see, it is possible to use it to transform it in a mode 'e'.

Is it not better to use a migration to avoid this ?